services/web/server/src/simcore_service_webserver/scicrunch/service_client.py

```diff
@@ -0,0 +1,189 @@
+        params={
+            "key": settings.api_key.get_secret_value(),
+            "field": "Resource Name",
+            "value": guess_name.strip(),
+        },
+        raise_for_status=True,
+    ) as resp:
+        body = await resp.json()
+        assert body.get("success")  # nosec
+        return ListOfResourceHits.parse_obj(body.get("data", []))
+
+
+class ValidationResult(IntEnum):
+    UNKNOWN = -1
+    INVALID = 0
+    VALID = 1
+
+
+class SciCrunchAPI:
+    """Client to communicate with scicrunch.org service
+
+    - wraps all calls to scicrunch.org API
+        - return result or raises web.HTTPError
+    - one instance per application
+        - uses app aiohttp client session instance
+        - uses settings
+    """
+
+    # FIXME: scicrunch timeouts should raise -> Service not avialable with a msg that scicrunch is currently not responding
+
+    RRID_RE = re.compile(RRID_PATTERN)
+
+    def __init__(self, client: ClientSession, settings: SciCrunchSettings):
+        self.settings = settings
+        self.client = client
+        self.base_url = URL.build(
+            scheme=self.settings.api_base_url.scheme,
+            host=self.settings.api_base_url.host,
+        )
+
+    # Website links ---------
+
+    def get_portal_link(self) -> str:
+        return str(self.base_url.with_path("/resources/"))
+
+    def get_rrid_link(self, rrid: str) -> str:
+        # NOTE: for some reason scicrunch query does not like prefix!
+        return str(
+            self.base_url.with_path("/resources/Any/search").with_query(
+                q="undefined", l=rrid.replace("RRID: ", "").strip()
+            )
+        )
+
+    def get_new_submission_link(self) -> str:
+        return str(self.base_url.with_path("/resources/about/resource"))
+
+    # Application instance ---------
+
+    @classmethod
+    def acquire_instance(
+        cls, app: MutableMapping[str, Any], settings: SciCrunchSettings
+    ) -> "SciCrunchAPI":
+        """ Returns single instance for the application and stores it """
+        obj = cls.get_instance(app)
+        if obj is None:
+            session = get_client_session(app)
+            app[f"{__name__}.SciCrunchAPI"] = obj = cls(session, settings)
+        return obj
+
+    @staticmethod
+    def get_instance(
+        app: MutableMapping[str, Any], *, raises=False
+    ) -> Optional["SciCrunchAPI"]:
+        """ Get's application instance """
+        obj = app.get(f"{__name__}.SciCrunchAPI")
+        if raises and obj is None:
+            raise web.HTTPServiceUnavailable(
+                reason="Link to scicrunch.org services is currently disabled."
+            )
+        return obj
+
+    # API calls --------
+
+    @classmethod
+    def validate_identifier(cls, rrid: str) -> str:
+        match = cls.RRID_RE.match(rrid.strip())
+        if match:
+            rrid = match.group(2)  # WARNING: captures indexing is 1-based
+            assert rrid and isinstance(rrid, str), "Captured {rrid}"  # nosec
```

Comment:
I would not use assert and replace them with if raise

Reply:
I think i mentioned it before ... but just to be sure
I only use assert for "defensive programming" while developing. I do not expect these asserts to be part of the running code in production (actually they are disabled). For that very reason is marked as ``# nosec``